packages/NodeCollector/BinaryOpConditionsCollector.php

```diff
@@ -23,6 +23,10 @@ final class BinaryOpConditionsCollector
      */
     public function findConditions(BinaryOp $binaryOp, string $binaryOpClass): array
     {
+        if ($binaryOp::class !== $binaryOpClass) {
```

Comment:
Alternately, I can make this an assertion but considering it a trivial case with zero operators feels more natural and also composes better with the next thing I got planned.